lua/lspconfig.lua

```diff
@@ -84,7 +86,7 @@ end
 local mt = {}
 function mt:__index(k)
   if configs[k] == nil then
-    pcall(require, 'lspconfig/'..k)
+    dofile(script_path .. 'lspconfig/' .. k .. ".lua")
```

Comment:
https://stackoverflow.com/a/31149198/152142 seems like there are some risks with dofile(), are we certain they are avoided here? Perhaps worth mentioning in a brief comment above this line.

Reply:
Yes, I'm not quite sure what "gotchas" there are with `dofile`, I believe the reasons it is faster are the drawbacks, it doesn't have to search package.path (thus requiring the debug workaround to get the path to the lspconfig.lua file), it doesn't cache lookups (which doesn't matter for us since we are only loading these files once to add to the global config table), and someone mentioned on IRC that it may avoid JIT compilation, which is fine because these are just config files.